src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,25 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.Arrays;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
```

Comment:
The `MachineProducer` interface should be parameterized to restrict its implementation to `Machine` types or its subtypes. The current implementation is correct as it uses `Excavator` which is a subtype of `Machine`.